kinds/preferences/preferences_kind.cue

```diff
@@ -44,5 +47,16 @@ lineage: schemas: [{
 			performance?: {}
 			functional?: {}
 		} @cuetsy(kind="interface")
+
+		#NavbarPreference: {
+			savedItems: [...#PreferenceNavLink]
+		} @cuetsy(kind="interface")
+
+		#PreferenceNavLink: {
+			id:     string
+			text:   string
+			url:    string
+			target: string
```

Comment:
Unfortunately I dont think we can do this, because we eventually will want to create a navigation `NavLink` out of this `PreferenceNavLink` and the `NavLink` does not have it as [optional](https://github.com/grafana/grafana/blob/main/pkg/services/navtree/models.go#L60), so the types fail when you try to assign one to the other 🤷